### PR TITLE
Add Git library resource type

### LINKED ISSUE #25: Resource: project libraries (Git references)
## Summary

Add a `libraries` resource type to manage DSS project library references (Git repositories).

## Motivation

DSS projects can import shared code from Git repositories as project libraries. These are used in Python/R recipes and notebooks. Managing library references as code ensures all environments point to the same repo, branch, and path.

## Suggested config

```yaml
libraries:
  - name: shared_utils
    repository: git@github.com:org/dss-shared-lib.git
    branch: main
    path: python
```

## Notes

- DSS supports multiple Git references per project.
- Libraries can specify sub-paths within the repo.
- Credentials/SSH keys for Git access are typically configured at the DSS instance level, not per-project.

### PR DESCRIPTION
**Pull request by jonas-meyer**
## Summary

- Add `dss_git_library` resource type with `GitLibraryResource` model and `GitLibraryHandler` CRUD handler
- Libraries are configured as a list under the `libraries:` key in YAML config
- Priority 10 ensures libraries are applied after variables (0) but before datasets/recipes (100)
- Handler uses DSS `DSSProjectGit` API: `add_library` (async clone), `set_library` + `reset_library` (sync update + async pull), `remove_library`

## Test plan

- [x] Resource model tests: address, defaults, custom values, extra forbid, model dump, plan priority
- [x] Handler tests: create, read, update, delete, edge cases (missing after create, python_path false)
- [x] Engine roundtrip: full create → noop → update → delete lifecycle
- [x] Priority ordering test: libraries planned before datasets
- [x] Config loader tests: YAML parsing, inclusion in resources, empty when omitted
- [x] Registry count updated (10 → 11)
- [x] Docs updated: resources, yaml-config, reference, architecture

Closes #25

### CODE REVIEW

File: src/dss_provisioner/engine/git_library_handler.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        git = self._get_git(ctx)
+        git.set_library(
+            git_reference_path=desired.name,
+            remote=desired.repository,
+            remotePath=desired.path,
+            checkout=desired.checkout,
+        )
+        future = git.reset_library(desired.name)
+        future.wait_for_result()
+        result = self._get_ref(ctx, desired.name)
```
[the remark this answers is not in the document]
Comment (jonas-meyer) [reply src/dss_provisioner/engine/git_library_handler.py]: Fixed — `update()` now checks `prior.attributes["add_to_python_path"]` and raises a clear `RuntimeError` if the value changed, instructing the user to delete and recreate.
